build.gradle.kts

```diff
@@ -0,0 +1,225 @@
+    compile(group = "com.google.code.gson", name = "gson", version = "2.8.1")
+
+    testCompile(group = "com.google.guava", name = "guava-testlib", version = "21.0")
+    testCompile("junit:junit:+")
+    testCompile("org.testfx:testfx-core:4.0.+")
+    testCompile("org.testfx:testfx-junit:4.0.+")
+    testRuntime(group = "org.testfx", name = "openjfx-monocle", version = "8u76-b04")
+}
+
+checkstyle {
+    configFile = file("$rootDir/checkstyle.xml")
+    toolVersion = "6.19"
+}
+
+pmd {
+    isConsoleOutput = true
+    sourceSets = setOf(java.sourceSets["main"], java.sourceSets["test"])
+    reportsDir = file("${project.buildDir}/reports/pmd")
+    ruleSetFiles = files(file("$rootDir/pmd-ruleset.xml"))
+    ruleSets = emptyList()
+}
+
+findbugs {
+    sourceSets = setOf(java.sourceSets["main"], java.sourceSets["test"])
+    excludeFilter = file("$rootDir/findBugsSuppressions.xml")
+    effort = "max"
+}
+
+fun printReportSafe(xmlReport: File) {
+    if (xmlReport.exists()) {
+        val bugs = (XmlParser().parse(xmlReport)["BugInstance"]) as Collection<*>
+        bugs.forEach {
+            println(XmlUtil.serialize(it as Node))
+        }
+    }
+}
+
+val findbugsMain: FindBugs by tasks
+val findbugsMainReport = task("findbugsMainReport") {
+    doLast {
+        printReportSafe(findbugsMain.reports.xml.destination)
+    }
+}
+val findbugsTest: FindBugs by tasks
+val findbugsTestReport = task("findBugsTestReport") {
+    doLast {
+        printReportSafe(findbugsTest.reports.xml.destination)
+    }
+}
+findbugsMain.finalizedBy(findbugsMainReport)
+findbugsTest.finalizedBy(findbugsTestReport)
+
+tasks.withType<JacocoReport> {
+    reports {
+        xml.isEnabled = true
+        html.isEnabled = true
+    }
+}
+
+tasks.withType<Test> {
+    testLogging {
+        if (project.hasProperty("logTests") || project.hasProperty("jenkinsBuild")) {
+            events("started", "passed", "skipped", "failed")
+        } else {
+            events("failed")
+        }
+        exceptionFormat = TestExceptionFormat.FULL
+    }
+    /*
+     * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
+     */
+    if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
+        println("Running UI Tests Headless")
+
+        jvmArgs = listOf(
+            "-Djava.awt.headless=true",
+            "-Dtestfx.robot=glass",
+            "-Dtestfx.headless=true",
+            "-Dprism.order=sw",
+            "-Dprism.text=t2k"
+        )
+        useJUnit {
+            this as JUnitOptions
+            excludeCategories("edu.wpi.first.shuffleboard.NonHeadlessTests")
+        }
+    }
+
+}
+
+application {
+    mainClassName = "edu.wpi.first.shuffleboard.Shuffleboard"
+}
+
+val sourceJar = task<Jar>("sourceJar") {
+    description = "Creates a JAR that contains the source code."
+    from(java.sourceSets["main"].allSource)
+    classifier = "sources"
+}
+
+tasks.withType<Jar> {
```

Comment:
There used to be a `classifier=null` here when this was done using the shadow plugin. I don't know why that was there. I don't think it was strictly necessary.